diff --git a/client/src/app/cart/page.tsx b/client/src/app/cart/page.tsx
index 094dbab..e9e0e1b 100644
--- a/client/src/app/cart/page.tsx
+++ b/client/src/app/cart/page.tsx
@@ -2,7 +2,7 @@
 
 import PaymentForm from "@/components/PaymentForm"
 import ShippingForm from "@/components/ShippingForm"
-import { CartItemsType } from "@/types"
+import { CartItemsType, ShippingFormInputs } from "@/types"
 import { ArrowRight, Trash2 } from "lucide-react"
 import Image from "next/image"
 import { useRouter, useSearchParams } from "next/navigation"
@@ -81,7 +81,7 @@ const Cartpage = () => {
   const searchParams=useSearchParams();
   const router=useRouter();
   const activeStep=parseInt(searchParams.get("step") || "1");
-  const [shippingForm,setShippingForm]=useState(null)
+  const [shippingForm,setShippingForm]=useState<ShippingFormInputs | null>(null)
   return (
     <div className="flex flex-col gap-8 items-center justify-center mt-12 ">
       {/* title */}
@@ -131,7 +131,7 @@ const Cartpage = () => {
             </div>
           ))
         ) : activeStep === 2 ? (
-          <ShippingForm />
+          <ShippingForm setShippingForm={setShippingForm}/>
         ) : activeStep === 3 ? (
           shippingForm ? (
             <PaymentForm />
diff --git a/client/src/components/ShippingForm.tsx b/client/src/components/ShippingForm.tsx
index 22656e3..9dc588b 100644
--- a/client/src/components/ShippingForm.tsx
+++ b/client/src/components/ShippingForm.tsx
@@ -1,7 +1,60 @@
+import { ShippingFormInputs, shippingFormSchema } from "@/types";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { ArrowRight } from "lucide-react";
+import { useRouter } from "next/navigation";
+import { SubmitHandler, useForm } from "react-hook-form"
 
-const ShippingForm = () => {
+const ShippingForm = ({setShippingForm}:{setShippingForm:(data:ShippingFormInputs)=>void}) => {
+  const {register,handleSubmit,formState:{errors}}=useForm<ShippingFormInputs>({
+    resolver:zodResolver(shippingFormSchema),
+  });
+  const router=useRouter();
+  const handleShippingForm:SubmitHandler<ShippingFormInputs>=(data)=>{
+    setShippingForm(data);
+    router.push("/cart?step=3",{scroll:false})
+  }
   return (
-    <div>ShippingForm</div>
+    <form className="flex flex-col gap-4 " onSubmit={handleSubmit(handleShippingForm)}>
+      <div className="flex flex-col gap-1">
+        <label className="text-xs text-gray font-medium " htmlFor="name">Name</label>
+        <input className="border-b border-gray-200 py-2 outline-none text-sm" type="text" id="name" placeholder="John Doe" {...register("name")} />
+        {errors.name && 
+          <p className="text-xs text-red-500">{errors.name.message}</p>
+        }
+      </div>
+      <div className="flex flex-col gap-1">
+        <label className="text-xs text-gray font-medium " htmlFor="email">Email</label>
+        <input className="border-b border-gray-200 py-2 outline-none text-sm" type="email" id="email" placeholder="JohnDoe@gmail.com" {...register("email")} />
+        {errors.email && 
+          <p className="text-xs text-red-500">{errors.email.message}</p>
+        }
+      </div>
+      <div className="flex flex-col gap-1">
+        <label className="text-xs text-gray font-medium " htmlFor="phone">Phone</label>
+        <input className="border-b border-gray-200 py-2 outline-none text-sm" type="text" id="phone" placeholder="987654321" {...register("phone")} />
+        {errors.phone && 
+          <p className="text-xs text-red-500">{errors.phone.message}</p>
+        }
+      </div>
+      <div className="flex flex-col gap-1">
+        <label className="text-xs text-gray font-medium " htmlFor="address">Address</label>
+        <input className="border-b border-gray-200 py-2 outline-none text-sm" type="text" id="address" placeholder="Civil Lines,NEW DELHI" {...register("address")} />
+        {errors.address && 
+          <p className="text-xs text-red-500">{errors.address.message}</p>
+        }
+      </div>
+      <div className="flex flex-col gap-1">
+        <label className="text-xs text-gray font-medium " htmlFor="city">City</label>
+        <input className="border-b border-gray-200 py-2 outline-none text-sm" type="text" id="city" placeholder="NEW DELHI" {...register("city")} />
+        {errors.city && 
+          <p className="text-xs text-red-500">{errors.city.message}</p>
+        }
+      </div>
+      <button type="submit" className="w-full bg-gray-800 hover:bg-gray-900 transition-all  text-white p-2 rounded-lg cursor-pointer flex items-center gap-2">
+              Continue
+              <ArrowRight className="w-3 h-3"/>
+            </button>
+    </form>
   )
 }
 
diff --git a/client/src/types.ts b/client/src/types.ts
index 5ab5d17..4cf3cc5 100644
--- a/client/src/types.ts
+++ b/client/src/types.ts
@@ -19,8 +19,8 @@ export type CartItemType=ProductType & {
 export type CartItemsType=CartItemType[]
 export const shippingFormSchema=z.object({
     name:z.string().min(1,"Name is required"),
-    email:z.email().min(1,"Email is required!"),
-    phone:z.email().min(7,"Phone number must be between 7 and 10 digits!").max(10,"Phone number must be between 7 and 10 digits!").regex(/^\d+$/,"Phone number must contain only numbers!"),
+    email:z.string().email().min(1,"Email is required!"),
+    phone:z.string().min(7,"Phone number must be between 7 and 10 digits!").max(10,"Phone number must be between 7 and 10 digits!").regex(/^\d+$/,"Phone number must contain only numbers!"),
     address:z.string().min(1,"Address is required!"),
     city:z.string().min(1,"City is required!"),
 })
